Mostrar imagens na busca de notícias

ekisu opened the pull request:
Agora as imagens das notícias são extraídas utilizando metadados das páginas pelo protocolo [OpenGraph](https://ogp.me/), e exibidas na tela de resultados:

![image](https://user-images.githubusercontent.com/5082637/68336831-fbb54000-00bd-11ea-8cc1-f88dfdeebce6.png)

ekisu committed:
{models, views}/news: adicionar suporte a imagens

Adiciona um campo, `image_url`, ao model News, e também adiciona suporte
para exibir tal campo no partial _news_card, usado na listagem de
notícias.

ekisu committed:
workers/news_feed_update_worker: extrair imagens antes de inserir

Usando um extrator de metadados do OpenGraph, agora o worker busca a
imagem antes de realizar a inserção no banco de dados.

Além disso, foi necessário mudar um pouco os testes dos workers, pois
a extração das imagens tornava os testes não repetíveis, adicionar um
mock ajuda a testar.

ekisu committed:
db/schema: arrumar conflito e aumentar versao

acontece :^)

ekisu committed:
views/news_card: melhorar aparencia dos cards

jimaeda reviewed (commented):
.

LucasPCBranco commented on features/noticias_mostrar_imagens.feature:
```diff
@@ -0,0 +1,11 @@
+    Para obter mais informações das notícias,
+    Gostaria de ver, na listagem das notícias, imagens presentes nelas.
+
+    Cenário: Busca de notícias
```
Peraí, agora a gente vai começar a fazer os testes em PT-BR? Fiquei confuso.

ekisu replied:
Ah, eu fiz só pra testar se funcionava, e esqueci de mudar - embora, por mim, faça mais sentido fazer os testes em português.

LucasPCBranco commented on spec/workers/update_all_sources_worker_spec.rb:
```diff
@@ -20,12 +20,12 @@
-        Sidekiq::Worker.drain_all # Isso força todos os workers a terminarem.
-
-        expect(News.count).to be > 0
+    it "should call NewsFeedUpdateWorker" do
```
Tem como explicar levemente o que está rolando aqui? Basicamente, é o negócio que força atualizar as notícias no Heroku?

ekisu replied:
Basicamente, eu testo se o `UpdateAllSourcesWorker` coloca jobs do tipo `NewsFeedUpdateWorker` na fila do Sidekiq (o nome do teste tá meio estranho, vou mudar isso depois).

É como se eu tivesse dado um mock no `NewsFeedUpdateWorker` pra testar se ele foi chamado, mas a maneira que o Sidekiq é usado dificulta usar mocks "normais".

LucasPCBranco reviewed (approved):
Ficou MUITO bom, velho, sério.

Eu fiz uns comentários isolados, que quando você puder dar uma olhadinha, sou eternamente grato.

ekisu committed:
config/database: aumentar numero padrão de conexões

Antes, o número padrão de conexões não era suficiente para comportar
as threads do Sidekiq e as conexões do Redis. Aumentar para uma quanti-
dade que seja suficiente.

ekisu committed:
spec/workers: melhorar documentação de um teste

Este teste estava com comentários/nome imprecisos, que não refletiam
o que ele realizava de verdade.
